pkg/storage/unified/README.md

```diff
@@ -781,3 +781,618 @@ The dual writer system provides metrics for monitoring:
+    )
+    
+    // Random load balancing within replication set
+    instance := replicationSet.Instances[rand.Intn(len(replicationSet.Instances))]
+    return instance.Id
+}
+```
+
+### Search Index Management
+
+Each search API server contains an embedded Bleve search engine that manages indexes for its assigned namespaces:
+
+```mermaid
+flowchart TD
+    subgraph "Search API Server"
+        A[Incoming Search Request] --> B{Index Exists?}
+        
+        B -->|Yes| C[Query Embedded Bleve Index]
+        B -->|No/Outdated| D[Just-in-Time Indexing]
+        
+        D --> E[Fetch Resources from Unified Storage]
+        E --> F[Build Search Documents]
+        F --> G{Index Size Threshold}
+        
+        G -->|Small < FileThreshold| H[Create Memory-based Index]
+        G -->|Large ≥ FileThreshold| I[Create Disk-based Index]
+        
+        H --> J[Store in Embedded Bleve]
+        I --> J
+        
+        J --> C
+        C --> K[Return Search Results]
+    end
+    
+    subgraph "Background Indexing"
+        L[Resource Write Events] --> M[Watch Event Stream]
+        M --> N[Index Queue Processor]
+        N --> O[Batch Document Updates]
+        O --> J
+    end
+    
+    subgraph "Index Features"
+        P[Full-text Search]
+        Q[Field-based Filtering]
+        R[Permission Filtering]
+        S[Usage Insights Integration]
+    end
+    
+    C --> P
+    C --> Q
+    C --> R
+    C --> S
+```
+
+#### Index Architecture Details:
+
+**Embedded Bleve Backend**: Each Search API Server contains its own Bleve search engine instance, not a shared external service.
+
+**Just-in-Time Indexing**: When a search request arrives for a namespace that doesn't have an index (or has an outdated index):
+1. The Search API Server fetches all resources for that namespace from Unified Storage
+2. Builds search documents in memory
+3. Creates either a memory-based or disk-based Bleve index depending on size
+4. Executes the search query against the newly built index
+5. Returns results to the user
+
+**Index Storage Strategy**:
+- **Memory indexes**: For small datasets (< `index_file_threshold` documents)
+- **Disk indexes**: For large datasets (≥ `index_file_threshold` documents)
+- Indexes are stored per Search API Server instance, not globally shared
+
+**Background Updates**: In addition to just-in-time indexing, Search API Servers also maintain indexes through background watch events for incremental updates.
+
+#### Index Configuration:
+```ini
+[unified_storage]
+; Path for disk-based search indexes
+index_path = /var/lib/grafana/unified-search/bleve
+
+; Threshold for file-based vs memory indexes  
+index_file_threshold = 1000
+
+; Maximum batch size for indexing
+index_max_batch_size = 100
+
+; Number of worker threads for indexing
+index_workers = 4
+
+; Cache TTL for indexes
+index_cache_ttl = 1h
+
+; Periodic rebuild interval (for usage insights)
+index_rebuild_interval = 24h
+```
```

Comment:
We could also include documentation about `index_min_count` and `index_max_count`.